operator/pkg/apis/controlplane/interface.go

```diff
@@ -0,0 +1,26 @@
+/*
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package controlplane
+
+import (
+	"context"
+
+	"github.com/awslabs/kit/operator/pkg/apis/controlplane/v1alpha1"
+)
+
+type ReconcileFinalize interface {
```

Comment:
I think this is classically called a Controller. Can we pull this interface out into something like pkg/controllers/types.go